src/Bladeburner/Skill.ts

```diff
@@ -35,9 +35,45 @@ export class Skill {
   }
 
   calculateCost(currentLevel: number, count = 1 as PositiveInteger): number {
-    return Math.floor(
-      count * this.baseCost * currentNodeMults.BladeburnerSkillCost +
-        this.costInc * currentNodeMults.BladeburnerSkillCost * (count * currentLevel + (count * (count - 1)) / 2),
+    /**
+     * The cost of the next level: (baseCost + currentLevel * costInc) * mult. The cost needs to be an integer, so we
+     * need to use Math.floor or Math.round.
+     *
+     * In order to calculate the cost of "count" levels, we need to run a loop. "count" can be a big number, so it's
+     * infeasible to calculate the cost in that way. We need to find the closed forms of:
+     *
+     * [1]:
+     * $$Cost = \sum_{i = CurrentLevel}^{CurrentLevel+Count-1}\lfloor ((BaseCost + i \ast CostInc) \ast Mult) \rfloor$$
+     *
+     * Or:
+     *
+     * [2]:
+     * $$Cost = \sum_{i = CurrentLevel}^{CurrentLevel+Count-1}\lceil ((BaseCost + i \ast CostInc) \ast Mult) \rceil$$
```

Comment:
Nice comprehensive comment! Only one nitpick: You're using `\lceil` here, which is for ceil(). AFAIK, there isn't a commonly accepted math symbol for round(), we just write the function out.

Reply:
I added a note for it. I can change `\lceil` and `\rceil` to the literal `Math.round` in the Tex expressions if you want.

Reply:
Good enough.